pkg/egressgateway/policy_test.go

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package egressgateway
+
+import (
+	"net/netip"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/types"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	"github.com/cilium/cilium/pkg/policy/api"
+)
+
+func TestPolicyConfig_updateMatchedEndpointIDs(t *testing.T) {
```

Comment:
One last thing, taking another look at this.  Although the change is pretty straightforward, I wonder if we should test the entire egw manager in `manager_privileged_test.go` to prevent regressions to this functionality.

Reply:
Done.